lib/postprocessing/image.js

```diff
@@ -16,10 +16,163 @@ const gm = require("./gm-promisify");
 const util = require('util');
 const exec = util.promisify(require('child_process').exec);
 
+// list of supported output formats
+// maps our fmt ('png') to imagemagick format ('PNG') https://imagemagick.org/script/formats.php
+const SUPPORTED_OUTPUT_FMT = {
+    "png": "PNG",
+    "jpg": "JPEG",
+    "jpeg": "JPEG",
+    "tif": "TIFF",
+    "tiff": "TIFF",
+    "gif": "GIF"
+};
+
+// list of supported input formats (intermediate rendition)
+// in imagemagick format https://imagemagick.org/script/formats.php
+const SUPPORTED_INPUT_FMT = new Set([
+    "PNG",
+    "JPEG",
+    "JPEG",
+    "TIFF",
+    "TIFF",
+    "GIF",
+    "BMP"
+]);
+
+// minimal instructions which many workers can already handle themselves
+// const BASIC_INSTRUCTIONS = new Set([
+//     "fmt",
+//     "width",
+//     "height",
+//     "quality"
+// ]);
+
+// list of post processing instructions we handle
+const EXTRA_INSTRUCTIONS = new Set([
+    "crop",
+    "jpegSize",
+    "interlace",
+    "dpi",
+    "convertToDpi"
+]);
+
+function isSupportedOutputFormat(instructions) {
+    return Object.keys(SUPPORTED_OUTPUT_FMT).includes(instructions.fmt);
+}
+
+function isSupportedInputFormat(metadata) {
+    return SUPPORTED_INPUT_FMT.has(metadata.format);
+}
+
+function hasExtraInstructions(instructions) {
+    return Object.keys(instructions).some(key => EXTRA_INSTRUCTIONS.has(key));
+}
+
+async function getImageMetadata(file) {
+    try {
+        // [1x1+0+0] is a trick to avoid reading all pixels into memory, improving time for large images
+        const result = await gm(`${file}[1x1+0+0]`).write("json:-");
+        const json = JSON.parse(result.toString());
+
+        // newer imagemagick returns an array (for multi-page formats), older just one object (6.8.9-9 at least)
+        if (Array.isArray(json)) {
+            return json[0].image;
+        } else {
+            return json.image;
+        }
+    } catch (e) {
+        throw new Error(`Reading metadata from intermediate rendition failed: ${e.message}`);
+    }
+}
+
+function isTargetFormat(metadata, instructions) {
+    const format = SUPPORTED_OUTPUT_FMT[instructions.fmt];
```

Comment:
`SUPPORTED_OUTPUT_FMT.has(instructions.fmt);` will do the same and use the Set API.